lib/kennel/models/record.rb

```diff
@@ -148,6 +148,52 @@ def invalid_update!(field, old_value, new_value)
 
       private
 
+      def validate_json_respecting_filters(data)
+        caught = begin
+                   validate_json(data)
+                   nil
+                 rescue ValidationError => e
+                   e
+                 end
+
+        if caught.nil?
+          unless skip_validations.empty?
+            invalid! nil, "skip_validations must not be used when no validations need skipping"
```

Comment:
this could backfire when used on a base monitor that others inherit from

Reply:
Yeah I'm rewriting this whole section. 

Essentially the concept which I'm aiming for, setting _specific_ validations to be skipped rather than all or none, is only of any value if a failing (but skipped) validation is not terminal.

In other words, given a record which has problems with both foo and bar, then

* report error foo
* report error bar

doesn't works as intended if "report error" is terminal (i.e. `raise`) – because if foo raised, then we found out that foo was skipped, then we never got to see bar.

So I'm currently working on changing the error reporting to be non-terminal (within each record), i.e. we run as many validations as possible then process them, including skipping, once all errors have been collected from all parts. This also has the arguable benefit of reporting all the available problems at once, instead of trickle-feeding (problem / fix it / different problem shows up / fix it / ... ).